api/v1alpha1/operator_types.go

```diff
@@ -37,6 +37,11 @@ type OperatorSpec struct {
 	//
 	// For more information on semver, please see https://semver.org/
 	Version string `json:"version,omitempty"`
+
+	//+kubebuilder:validation:MaxLength:=48
+	//+kubebuilder:validation:Pattern:=^[a-z0-9]+(-[a-z0-9]+)*$
```

Comment:
+1 on a super-well-defined pattern for channel name. Not important for this particular PR, but we should probably make sure `opm` agrees with this pattern (e.g. as part of `opm validate`)